screen dislocation (OSX)

My Mac version: 10.13.3

When I execute `  cargo run --release --features "winit glium" --example all_winit_glium`, it was successfully compiled.

However ,as shown in Fig. below, it is not as expected.

![1518444409370](https://user-images.githubusercontent.com/33049497/36100578-ad394eee-1041-11e8-8b3e-89f7074ea98f.jpg)

That problem is now fixed in winit 0.12 but there's an incompatible change somewhere in event conversion.

```rust
error[E0308]: mismatched types
  --> examples/text.rs:73:76
   |
73 |                 if let Some(event) = conrod::backend::winit::convert_event(event.clone(), &display) {
   |                                                                            ^^^^^^^^^^^^^ expected enum `winit::events::Event`, found enum `conrod::<unnamed>::<unnamed>::Event`
   |
   = note: expected type `winit::events::Event`
              found type `conrod::<unnamed>::<unnamed>::Event`
note: Perhaps two different versions of crate `winit` are being used?
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -48,8 +48,8 @@ rusttype = { version = "0.5.0", features = ["gpu_cache"] }
 # - Converting piston `GenericEvent` types to `conrod::event::Raw`s.
 # - Rendering the `conrod::render::Primitives` yielded by `Ui::draw`.
 # Enables the `conrod::backend::piston` module.
-glium = { version = "0.20", optional = true }
-winit = { version = "0.10", optional = true }
+winit = { version = "0.12", optional = true }
+glium = { version = "0.21", optional = true }
 piston2d-graphics = { version = "0.24", optional = true }
 gfx = { version = "0.17", optional = true }
 gfx_core = { version = "0.8", optional = true }
@@ -65,6 +65,6 @@ petgraph = "0.4"
 rand = "0.4"
 # glutin_gfx.rs example dependencies
 gfx_window_glutin = "0.20"
-glutin = "0.12"
+glutin = "0.14"
 # piston_window.rs example dependencies
 piston_window = "0.75"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -64,7 +64,7 @@ image = "0.18"
 petgraph = "0.4"
 rand = "0.4"
 # glutin_gfx.rs example dependencies
-gfx_window_glutin = "0.20"
+gfx_window_glutin = "0.22"
 glutin = "0.14"
 # piston_window.rs example dependencies
 piston_window = "0.75"

--- a/src/backend/winit.rs
+++ b/src/backend/winit.rs
@@ -16,7 +16,7 @@ use cursor;
 /// This allows users to pass either `glium::Display`, `glium::glutin::Window` or `winit::Window`
 /// to the `conrod::backend::winit::convert` function defined below.
 pub trait WinitWindow {
-    /// Return the inner size of the window.
+    /// Return the inner size of the window in pixels.
     fn get_inner_size(&self) -> Option<(u32, u32)>;
     /// Return the window's DPI factor so that we can convert from pixel values to scalar values.
     fn hidpi_factor(&self) -> f32;
@@ -60,15 +60,15 @@ pub fn convert_event<W>(e: winit::Event, window: &W) -> Option<Input>
 pub fn convert_window_event<W>(e: winit::WindowEvent, window: &W) -> Option<Input>
     where W: WinitWindow,
 {
-    // The window size in points.
-    let (win_w, win_h) = match window.get_inner_size() {
-        Some((w, h)) => (w as Scalar, h as Scalar),
-        None => return None,
-    };
-
     // The "dots per inch" factor. Multiplying this by `win_w` and `win_h` gives the framebuffer
     // width and height.
     let dpi_factor = window.hidpi_factor() as Scalar;
+
+    // The window size in points.
+    let (win_w, win_h) = match window.get_inner_size() {
+        Some((w, h)) => (w as Scalar / dpi_factor, h as Scalar / dpi_factor),
+        None => return None,
+    };
 
     // Translate the coordinates from top-left-origin-with-y-down to centre-origin-with-y-up.
     //

